### DISCUSSION TITLE
how to disable camel casing in Elsa server

### DISCUSSION THREAD

**Discussion by akghasemi** [elsa-workflows/elsa-core#2493] [Q&A]
hi, I use the following code in startup to disable camel casing by Elsa but it is not working

                var apiOption= new Elsa.Server.Api.ElsaApiOptions();

                apiOption.SetupNewtonsoftJson=new Action<MvcNewtonsoftJsonOptions>(c=>c.UseCamelCasing(false));
                services.AddElsaApiEndpoints(apiOption);
am I missing something here?

**Comment by sfmskywalker** [comment]
Yes, make sure to also configure ASP.NET's JSON settings - some API endpoints use these settings.

Example: 

```csharp
services.AddControllers().AddNewtonsoftJson(json => json.SerializerSettings.ContractResolver = new DefaultContractResolver());
```

**Reply by ricksearcy** [reply]
@sfmskywalker can you explain how this is done in V3 server? 

I have tried below and it does not work. I can't find any Elsa specific options in V3. 

 services.AddControllers()
       .AddJsonOptions(options =>
       {
           options.JsonSerializerOptions.PropertyNamingPolicy = null;
       }).AddNewtonsoftJson(options =>
       {
           options.SerializerSettings.ContractResolver = new DefaultContractResolver();
       }); 
        services.AddMvc()
    .AddJsonOptions(options => options.JsonSerializerOptions.PropertyNamingPolicy = null);

**Reply by sfmskywalker** [reply]
Hi @ricksearcy,

In Elsa 3, we've transitioned to using FastEndpoints for API configurations. The request and response serialization is primarily set up within the `MapWorkflowsApi` method. To provide more flexibility for application-level customizations, we can consider introducing an additional parameter. This would be a callback function that receives the FastEndpoint options, allowing developers like you to modify settings, such as property name casing.

If you're also utilizing Elsa Studio, it's crucial to synchronize its Elsa.Api.Client's Refit serialization settings with any changes you make. We recognize the need to make these settings more accessible and are looking into ways to expose them for easier configuration.

If you're inclined, please feel free submit a PR with the proposed changes. Alternatively, you can open an issue detailing your suggestions, and we'll ensure it's reviewed and added to our development roadmap.

Thank you for your feedback and engagement with Elsa!
